catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveCatalogPropertiesMeta.java

```diff
@@ -36,6 +36,10 @@ public class HiveCatalogPropertiesMeta extends BaseCatalogPropertiesMetadata {
 
   public static final String FETCH_TIMEOUT_SEC = "kerberos.keytab-fetch-timeout-sec";
 
+  public static final String LIST_ALL_TABLES = "list-all-tables";
+
+  public static final boolean DEFAULT_LIST_ALL_TABLES = false;
```

Comment:
Due to [potential performance issues](https://github.com/apache/iceberg/issues/3907) that table filters may introduce when there are many tables, we should set the default value to `true`, and the default `true` also align with Hive client behavior.

Reply:
I took @FANNG1 advice ，set this to be fase。so,  @FANNG1 how do you think this advice?

Reply:
talked with @mchades,  we could remove the `list-all-tables` parameters and the default action is list only hive tables because we already need to get table meta data when listing table for now. If encountering performance issues we could add a parameter to control whether get all tables, but we could do track it in another issue.